pay-api/src/pay_api/services/eft_short_name_historical.py

```diff
@@ -0,0 +1,194 @@
+            short_name_id=history.short_name_id,
+            transaction_date=EFTShortnameHistorical.transaction_date_now(),
+            transaction_type=EFTHistoricalTypes.FUNDS_RECEIVED.value
+        )
+
+    @staticmethod
+    @user_context
+    def create_statement_paid(history: EFTShortnameHistory, **kwargs) -> EFTShortnamesHistoricalModel:
+        """Create EFT Short name statement paid historical record."""
+        return EFTShortnamesHistoricalModel(
+            amount=history.amount,
+            created_by=kwargs['user'].user_name,
+            credit_balance=history.credit_balance,
+            description='Statement Paid',
+            hidden=history.hidden,
+            is_processing=history.is_processing,
+            payment_account_id=history.payment_account_id,
+            related_group_link_id=history.related_group_link_id,
+            short_name_id=history.short_name_id,
+            statement_number=history.statement_number,
+            transaction_date=EFTShortnameHistorical.transaction_date_now(),
+            transaction_type=EFTHistoricalTypes.STATEMENT_PAID.value
+        )
+
+    @staticmethod
+    @user_context
+    def create_statement_reverse(history: EFTShortnameHistory, **kwargs) -> EFTShortnamesHistoricalModel:
+        """Create EFT Short name statement reverse historical record."""
+        return EFTShortnamesHistoricalModel(
+            amount=history.amount,
+            created_by=kwargs['user'].user_name,
+            credit_balance=history.credit_balance,
+            description='Payment Reversed',
+            hidden=history.hidden,
+            is_processing=history.is_processing,
+            payment_account_id=history.payment_account_id,
+            related_group_link_id=history.related_group_link_id,
+            short_name_id=history.short_name_id,
+            statement_number=history.statement_number,
+            transaction_date=EFTShortnameHistorical.transaction_date_now(),
+            transaction_type=EFTHistoricalTypes.STATEMENT_REVERSE.value
+        )
+
+    @staticmethod
+    def transaction_date_now() -> datetime:
+        """Construct a datetime using the legislation timezone."""
+        return datetime.now(tz=timezone.utc)
+
+    @staticmethod
+    def _get_account_name():
+        """Return case statement for deriving payment account name."""
+        return case(
+            (PaymentAccountModel.name.like('%-' + PaymentAccountModel.branch_name),
+             func.replace(PaymentAccountModel.name, '-' + PaymentAccountModel.branch_name, '')
+             ), else_=PaymentAccountModel.name).label('account_name')
+
+    @classmethod
+    def search(cls, short_name_id: int,
+               search_criteria: EFTShortnameHistorySearch = EFTShortnameHistorySearch()):
+        """Return EFT Short name history by search criteria."""
+        history_model = aliased(EFTShortnamesHistoricalModel)
+        latest_history_model = aliased(EFTShortnamesHistoricalModel)
```

Comment:
I think I was looking for this instead of subquery (in the short name linking search route).. will keep in mind for the future